### PR TITLE
feat(eval): add eval-run-challenger.py for Sovereign AI Eval Phase 2

### PR DESCRIPTION
**Pull request by TimeToBuildBob**
## Summary
- Phase 2 of the Sovereign AI Model Evaluation Harness (idea #567): `scripts/eval-run-challenger.py` replays a Phase-1 trace's oracle input (first user message extracted from its trajectory via `gptme_sessions.transcript`) against a challenger model in an isolated worktree.
- Records behavioral metrics — commits-made match, duration ratio, outcome match — comparing the challenger's behavior to the oracle session's recorded envelope, and renders a per-category markdown comparison table.
- `--dry-run` mode extracts and validates oracle inputs without spawning any model calls or requiring a worktree — verified end-to-end against the real trace file (`state/eval-traces/traces-2026-06-22.jsonl` in the brain repo): 2/3 sampled traces had a usable first user message.
- Adds `gptme_sessions` to `mypy.ini`'s `ignore_missing_imports` list (same pattern as `gptme_subscription`/`gptme_usage`) since the script imports it via `sys.path.insert` rather than as an installed package.

## Scope
- In scope: trace loading/filtering, oracle-input extraction, challenger invocation + git-diff-based metric collection, comparison report.
- Out of scope (follow-up): LLM-judge scoring integration (`judge_session` exists and is reusable, left as a `--llm-judge` extension point), `files_changed_jaccard` (needs the oracle's exact changed-file list, not currently in the trace schema), Thompson-bandit feedback wiring — all explicitly marked optional/Phase-3 in the design doc.

## Test plan
- [x] `uv run pytest tests/test_eval_run_challenger.py -q` — 9/9 passing (trace loading/filtering, oracle-input extraction incl. no-user-message case, metric computation incl. zero-duration edge case, report rendering)
- [x] `ruff check` / `ruff format --check` clean
- [x] `prek run mypy --files scripts/eval-run-challenger.py tests/test_eval_run_challenger.py mypy.ini` — passing
- [x] Live `--dry-run` against real Phase-1 trace data confirms the extraction pipeline works end-to-end

### CODE REVIEW

**Comment by TimeToBuildBob** [comment]
Addressed Greptile P2: oracle_input now written to a tempfile and read via shell command substitution (`$(cat file)`) instead of being passed directly in `argv`. This avoids `E2BIG` / `ARG_MAX` issues for very long prompts embedded in production traces.

Tested: 9/9 tests pass, `ruff check` clean. Pushed to branch.

**Comment by TimeToBuildBob** [comment]
@greptileai review

File: scripts/eval-run-challenger.py [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+            )
+            continue
+
+        challenger = run_challenger(
+            oracle_input,
+            worktree=args.brain_worktree,
+            challenger_cmd=args.challenger_cmd,
+            timeout=args.timeout,
+        )
+        metrics = compute_metrics(trace, challenger)
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply scripts/eval-run-challenger.py:339]: Fixed: the loop now captures the worktree's HEAD once before iterating and hard-resets + cleans back to it before each trace (`_git_reset_hard`, scripts/eval-run-challenger.py:129 + main() loop). Added a unit test (`test_git_reset_hard_discards_commits_and_untracked_files`) that simulates the accumulation bug and verifies the reset restores a clean baseline. Pushed in 14e8b87.
